compositor_render/src/renderer.rs

```diff
@@ -66,15 +73,23 @@ pub struct RenderCtx<'a> {
 }
 
 pub struct RegisterCtx {
+    pub renderer_id: Arc<str>,
     pub wgpu_ctx: Arc<WgpuCtx>,
     pub chromium: Arc<ChromiumContext>,
 }
 
 impl Renderer {
     pub fn new(opts: RendererOptions) -> Result<Self, InitRendererEngineError> {
         let wgpu_ctx = Arc::new(WgpuCtx::new()?);
+        let renderer_id = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(30)
+            .map(char::from)
+            .collect::<String>()
+            .into();
```

Comment:
[NIT] IMO it's more readable with separate function
```suggestion
        let renderer_id = rand_str()
```